Apple ARM64 debug build broken

```
    LINK build/dev/usr/lib/libjulia-internal-debug.1.9.dylib
Undefined symbols for architecture arm64:
  "_ptrauth_strip", referenced from:
      _jl_rec_backtrace in stackwalk.dbg.obj
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Caused by https://github.com/JuliaLang/julia/pull/45110.

--- a/src/stackwalk.c
+++ b/src/stackwalk.c
@@ -789,7 +789,7 @@ _os_tsd_get_direct(unsigned long slot)
 // Unconditionally defined ptrauth_strip (instead of using the ptrauth.h header)
 // since libsystem will likely be compiled with -mbranch-protection, and we currently are not.
 // code from https://github.com/llvm/llvm-project/blob/7714e0317520207572168388f22012dd9e152e9e/compiler-rt/lib/sanitizer_common/sanitizer_ptrauth.h
-inline uint64_t ptrauth_strip(uint64_t __value, unsigned int __key) {
+static inline uint64_t ptrauth_strip(uint64_t __value, unsigned int __key) {
   // On the stack the link register is protected with Pointer
   // Authentication Code when compiled with -mbranch-protection.
   // Let's strip the PAC unconditionally because xpaclri is in the NOP space,

